src/main/java/uk/gov/hmcts/ccd/domain/service/common/LegacyCaseDataAccessControl.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.caseaccess.CachedCaseUserRepository;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseUserRepository;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.service.AccessControl;
+import uk.gov.hmcts.ccd.infrastructure.user.UserAuthorisation;
+
+import java.util.Optional;
+
+import static uk.gov.hmcts.ccd.data.caseaccess.GlobalCaseRole.CREATOR;
+
+/**
+ * Delete this class after we migrate CaseDataAccessControl fully.
+ */
+@Deprecated
+@Component
+@ConditionalOnProperty(name = "ccd.new-access-control-enabled", havingValue = "false", matchIfMissing = true)
+public class LegacyCaseDataAccessControl implements CaseDataAccessControl, AccessControl {
```

Comment:
Instead of using Legacy, can we rename this class to RoleBasedAccessControl

Reply:
How about RoleBasedCaseDataAccessControl?

Reply:
`RoleBasedCaseDataAccessControl` is good for me